### ISSUE TITLE
Compilation Error on Mac OS X 10.10

### ISSUE THREAD

**Issue by ayberk** [ContinuumIO/nutchpy#20]
I'm getting an compilation error while trying to build the latest version. Here's the output:
```
➜  nutchpy git:(master) sudo python setup.py install
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building seqreader-app 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ seqreader-app ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /Users/ayberk/nutchpy/seqreader-app/src/main/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ seqreader-app ---
[INFO] Changes detected - recompiling the module!
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Compiling 6 source files to /Users/ayberk/nutchpy/seqreader-app/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/ayberk/nutchpy/seqreader-app/src/main/java/com/continuumio/seqreaderapp/RecordIterator.java:[19,8] com.continuumio.seqreaderapp.RecordIterator is not abstract and does not override abstract method remove() in java.util.Iterator
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.077 s
[INFO] Finished at: 2015-09-29T17:16:43-07:00
[INFO] Final Memory: 17M/115M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project seqreader-app: Compilation failure
[ERROR] /Users/ayberk/nutchpy/seqreader-app/src/main/java/com/continuumio/seqreaderapp/RecordIterator.java:[19,8] com.continuumio.seqreaderapp.RecordIterator is not abstract and does not override abstract method remove() in java.util.Iterator
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Traceback (most recent call last):
  File "setup.py", line 132, in <module>
    shutil.copy(jar_file,java_lib_dir)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 119, in copy
    copyfile(src, dst)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 82, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 2] No such file or directory: 'seqreader-app/target/seqreader-app-1.0-SNAPSHOT-jar-with-dependencies.jar'
```

I use java version "1.7.0_79" and python version "2.7.10".

**Comment by thammegowda** [comment]
@ayberk Please upgrade your build environment to JDK8 to get started immediately.

**Comment by ahmadia** [comment]
fixed in #20
